nima/tests/integration/webclient/webclient/src/test/java/io/helidon/nima/tests/integration/webclient/AuthHttpProxyTest.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright (c) 2023 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.nima.tests.integration.webclient;
+
+import static io.helidon.common.http.Http.Method.GET;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.jupiter.api.Assertions.fail;
+
+import io.helidon.common.http.Http;
+import io.helidon.nima.http2.webclient.Http2Client;
+import io.helidon.nima.testing.junit5.webserver.ServerTest;
+import io.helidon.nima.testing.junit5.webserver.SetUpRoute;
+import io.helidon.nima.webclient.api.HttpClient;
+import io.helidon.nima.webclient.api.HttpClientResponse;
+import io.helidon.nima.webclient.api.Proxy;
+import io.helidon.nima.webclient.api.Proxy.ProxyType;
+import io.helidon.nima.webclient.http1.Http1Client;
+import io.helidon.nima.webserver.WebServer;
+import io.helidon.nima.webserver.http.HttpRouting;
+
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+@ServerTest
+class AuthHttpProxyTest {
+
+    private static final String PROXY_HOST = "localhost";
+    private static final String USER = "user";
+    private static final String PASSWORD = "password";
+    private int proxyPort;
+    private HttpProxy httpProxy;
+
+    private final HttpClient<?> clientHttp1;
+    private final HttpClient<?> clientHttp2;
+    
+    @SetUpRoute
+    static void routing(HttpRouting.Builder router) {
+        router.route(GET, "/get", Routes::get);
+    }
+
+    @BeforeEach
+    void before() {
+        httpProxy = new HttpProxy(0, USER, PASSWORD);
+        httpProxy.start();
+        proxyPort = httpProxy.connectedPort();
+        assertThat(httpProxy.counter(), is(0));
+    }
+
+    @AfterEach
+    void after() {
+        httpProxy.stop();
+    }
+
+    AuthHttpProxyTest(WebServer server) {
```

Comment:
You can just inject the clients into constructor parameters:
```
AuthHttpProxyTest(Http1Client http1Client, Http2Client http2Client)
```
and they will be initialized to the host and port of the server

Reply:
I am getting:
```
[ERROR] io.helidon.nima.tests.integration.webclient.AuthHttpProxyTest.testUserPasswordNotCorrect1  Time elapsed: 0 s  <<< ERROR!
org.junit.jupiter.api.extension.ParameterResolutionException: No ParameterResolver registered for parameter [io.helidon.nima.http2.webclient.Http2Client arg1] in constructor [io.helidon.nima.tests.integration.webclient.AuthHttpProxyTest(io.helidon.nima.webclient.http1.Http1Client,io.helidon.nima.http2.webclient.Http2Client)].
	at org.junit.jupiter.engine.execution.ParameterResolutionUtils.resolveParameter(ParameterResolutionUtils.java:120)
	at org.junit.jupiter.engine.execution.ParameterResolutionUtils.resolveParameters(ParameterResolutionUtils.java:103)
	at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker.invoke(InterceptingExecutableInvoker.java:59)
	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.invokeTestClassConstructor(ClassBasedTestDescriptor.java:363)

```

Reply:
please add the `http2` testing support as well

Reply:
```xml
        <dependency>
            <groupId>io.helidon.nima.testing.junit5</groupId>
            <artifactId>helidon-nima-testing-junit5-http2</artifactId>
            <scope>test</scope>
        </dependency>
```